java/src/com/ibm/streamsx/topology/internal/context/BuildServiceCompile.java

```diff
@@ -0,0 +1,145 @@
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Path;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+
+import org.apache.http.HttpEntity;
+import org.apache.http.HttpStatus;
+import org.apache.http.client.ClientProtocolException;
+import org.apache.http.client.HttpClient;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.client.methods.HttpRequestBase;
+import org.apache.http.entity.ContentType;
+import org.apache.http.entity.mime.MultipartEntityBuilder;
+import org.apache.http.entity.mime.content.FileBody;
+import org.apache.http.entity.mime.content.StringBody;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.apache.http.util.EntityUtils;
+
+import com.ibm.json.java.JSON;
+import com.ibm.json.java.JSONObject;
+import com.ibm.json.java.OrderedJSONObject;
+import com.ibm.streamsx.topology.Topology;
+import com.ibm.streamsx.topology.internal.process.CompletedFuture;
+
+public class BuildServiceCompile extends ZippedToolkitStreamsContext {
+
+	@Override
+	public Future<File> submit(Topology app, Map<String, Object> config) throws Exception {  
+		String namespace = (String) app.builder().json().get("namespace");
+		String mainComposite = (String) app.builder().json().get("name");
+		
+        File zippedArchive = super.submit(app, config).get();
+        
+        File bundle = doRemoteCompile(namespace, mainComposite, zippedArchive);
+        
+        return new CompletedFuture<File>(bundle);
+	}
+	
+	@Override
+	public Future<File> submit(JSONObject submission) throws InterruptedException, ExecutionException, Exception{
+		JSONObject jsonGraph = (JSONObject) submission.get(SUBMISSION_GRAPH);
+		String namespace = jsonGraph.get("namespace").toString();
+		String mainComposite = jsonGraph.get("name").toString();
+		
+		File zippedArchive = super.submit(submission).get();
+		File bundle = doRemoteCompile(namespace, mainComposite, zippedArchive);
+
+		return new CompletedFuture<File>(bundle);
+		
+	}
+	
+	private File doRemoteCompile(String namespace, String mainComposite, 
+			File zippedArchive) throws ClientProtocolException, IOException {
+		String bundleName = namespace + "." + mainComposite + ".sab";
+		String archiveName = zippedArchive.getName();
+		HttpClient client = HttpClientBuilder.create().build();
+		
+		File archiveFile = new File(archiveName);
+		HttpPost postJobWithConfig = new HttpPost(getBuildFarmURL());
+        postJobWithConfig.addHeader("accept", ContentType.APPLICATION_JSON.getMimeType());
+
+        FileBody bundleBody = new FileBody(archiveFile, 
+        		ContentType.APPLICATION_OCTET_STREAM, archiveName);
+        
+        StringBody configBody = new StringBody(createBuildConfig(namespace, mainComposite),
```

Comment:
application/json was the first MIME type I tried, but the text/default was the only setting which worked successfully.